packages/falter-berlin-migration/uci-defaults/freifunk-berlin-01-migration.sh

```diff
@@ -661,6 +669,33 @@ r1_1_1_rssiled() {
   config_foreach handle_wifi_iface_rssiled wifi-iface
 }
 
+r1_1_2_bump_repo() {
+  # adjust the opkg packagefeed to point to new version
+  log "bumping packagefeed to 1.1.2"
+  sed -ie 's|firmware.berlin.freifunk.net/feed/.*/packages|firmware.berlin.freifunk.net/feed/1.1.2/packages|g' /etc/opkg/customfeeds.conf
```

Comment:
I think we will have to repeat this for each new version.  Would it be possible to generalize the function and pass the $VERSION as an arguement?  Another way would be to pul the "firmware.berlin.freifunk.net/feed" line from /rom/etc/opkg/customfeeds.conf and replace it in the upgraded customfeeds.conf (preserving any user-added custom feeds is very important).

Reply:
solved in new version.

Reply:
You changed the code to do the following:
```sed -n 's|.\+\([0-9]\.[0-9]\.[0-9].\?\)\/.\+$|\1|p' /rom/etc/opkg/customfeeds.conf```

this may not always work.  How about searching the file for ```src/gz openwrt_falter``` instead.  Then we know that the desired repo will be found no matter what the feed looks like.

For example, there is the feed directory https://firmware.berlin.freifunk.net/feed/snapshot/ which would fail with the above code.

Reply:
I agree on the `src/gz openwrt_falter` matter, but don't understand the example. We don't support migration on snapshot builds, do we?

Reply:
@pmelange I added your suggestion with the feed name. My assumption was first, that `?** would work non-greedy in sed, but it doesn't. I have to investigate this further for the right regex. I'd like to match stuff like `1.1.2-pre` too.

Reply:
> I agree on the `src/gz openwrt_falter` matter, but don't understand the example. We don't support migration on snapshot builds, do we?

I don't really think of this bit of code as a migration.  Instead I think of it as keeping the package feed in alignment with the installed image.  And yes, changes to /etc/config are not supported in snapshot builds.

Reply:
> @pmelange I added your suggestion with the feed name. My assumption was first, that `?** would work non-greedy in sed, but it doesn't. I have to investigate this further for the right regex. I'd like to match stuff like `1.1.2-pre` too.

I don't think that it is very necessary for us to pull the $FEED_VERSION from /rom/etc/opkg/customfeeds.conf.  Why not just take the entire line and replace it in the overlay system.

There may be an edge-case where a user has modified the customfeeds.conf file.  In this case the user could have commented out the original feed and added a new one for ```openwrt_falter```.  So it might make sense to check for any leading ```#```'s before going ahead a replacing anything.

Reply:
right, this should be solved in the recent version now. It changes the repo line only, if it matches our feedname plus the host-name the feed is located on. Thus any customizations should be preserved, but commented out original lines will be migrated as well.

Reply:
That looks great!